### PR TITLE
OCPNODE-1890: Bump k8s api to v0.29.0

### PR DESCRIPTION
**Pull request by sairameshv**
cc: @soltysh

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/cluster-kube-apiserver-operator
 
-go 1.20
+go 1.21.3
```
Comment (soltysh) [inline go.mod:3]: Same as before go1.20

**Review by soltysh — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (sairameshv) [reply go.mod]: Addressed

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**